Front-end/store/src/app/core/services/telegram-auth.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+import { Route, Router } from '@angular/router';
+
+import { PROFILE_URL, TELEGRAM_URL } from '../utils/values';
+
+import { UserProfile } from '../models/user-profile';
+
+import { AuthorizationService } from './authorization.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TelegramAuthService {
```

Comment:
Maybe TelegramBotContactService or TelegramBotLinkService  because it is not authorization service.